restdocs-raml/src/main/java/com/epages/restdocs/raml/RamlDocumentation.java

```diff
@@ -0,0 +1,141 @@
+        }
+
+        return MockMvcRestDocumentation.document(identifier, enhancedSnippets);
+    }
+
+    protected static Snippet[] enhanceSnippetsWithRaml(String description,
+                                                       boolean privateResource,
+                                                       Function<List<Snippet>, List<Snippet>> snippetFilter,
+                                                       Snippet... snippets) {
+
+        List<RequestFieldsSnippet> requestFieldsSnippets = new ArrayList<>();
+        List<ResponseFieldsSnippet> responseFieldsSnippets = new ArrayList<>();
+        List<LinksSnippet> linkSnippets = new ArrayList<>();
+        List<RequestParametersSnippet> requestParametersSnippets = new ArrayList<>();
+
+        List<Snippet> ramlSnippets = new ArrayList<>();
+
+        List<Snippet> otherSnippets = new ArrayList<>();
+
+        for (Snippet snippet : snippets) {
+            if (snippet instanceof RequestFieldsSnippet) {
+                requestFieldsSnippets.add((RequestFieldsSnippet) snippet);
+            } else if (snippet instanceof ResponseFieldsSnippet) {
+                responseFieldsSnippets.add((ResponseFieldsSnippet) snippet);
+            } else if (snippet instanceof LinksSnippet) {
+                linkSnippets.add((LinksSnippet) snippet);
+            } else if (snippet instanceof RequestParametersSnippet) {
+                requestParametersSnippets.add((RequestParametersSnippet) snippet);
+            } else if (snippet instanceof RamlResourceSnippet) {
+                ramlSnippets.add(snippet);
+            } else {
+                otherSnippets.add(snippet);
+            }
+        }
+
+        List<Snippet> enhancedSnippets = new ArrayList<>();
+        enhancedSnippets.addAll(requestFieldsSnippets);
+        enhancedSnippets.addAll(responseFieldsSnippets);
+        enhancedSnippets.addAll(linkSnippets);
+        enhancedSnippets.addAll(requestParametersSnippets);
+        enhancedSnippets.addAll(otherSnippets);
+
+        if (ramlSnippets.isEmpty()) { // No RamlResourceSnippet, so we configure our own based on the info of the other snippets
+            RamlResourceSnippetParameters ramlParameters = RamlResourceSnippetParameters.builder()
+                    .description(description)
+                    .privateResource(privateResource)
+                    .requestFields(requestFieldsSnippets.stream().map(DescriptorExtractor::extract).flatMap(List::stream).toArray(FieldDescriptor[]::new))
+                    .responseFields(responseFieldsSnippets.stream().map(DescriptorExtractor::extract).flatMap(List::stream).toArray(FieldDescriptor[]::new))
+                    .links(linkSnippets.stream().map(DescriptorExtractor::extract).flatMap(List::stream).toArray(LinkDescriptor[]::new))
+                    .requestParameters(requestParametersSnippets.stream().map(DescriptorExtractor::extract).flatMap(List::stream).toArray(ParameterDescriptor[]::new))
```

Comment:
we should also handle `PathParametersSnippet`

Reply:
Done!